ui/src/containers/preferences.list/details.tsx

```diff
@@ -0,0 +1,206 @@
+import PageHeader from "~/components/page-header";
+import {
+  Button,
+  Flex,
+  Grid,
+  Separator,
+  Switch,
+  Text,
+  TextField,
+} from "@raystack/apsara";
+import { useCallback, useEffect, useState } from "react";
+import { V1Beta1Preference, V1Beta1PreferenceTrait } from "@raystack/frontier";
+import { useParams } from "react-router-dom";
+import { useFrontier } from "@raystack/frontier/react";
+import Skeleton from "react-loading-skeleton";
+import dayjs from "dayjs";
+import * as R from "ramda";
+import { toast } from "sonner";
+
+interface PreferenceValueProps {
+  trait: V1Beta1PreferenceTrait;
+  value: string;
+  onChange: (v: string) => void;
+}
+
+function PreferenceValue({ value, trait, onChange }: PreferenceValueProps) {
+  if (R.has("checkbox")(trait)) {
+    const checked = value === "true";
+    return (
+      <Switch
+        // @ts-ignore
+        checked={checked}
+        onCheckedChange={(v: boolean) => onChange(v.toString())}
+      />
+    );
+  } else if (R.or(R.has("text"), R.has("textarea"))(trait)) {
+    return (
+      <TextField value={value} onChange={(e) => onChange(e.target.value)} />
+    );
+  } else {
+    return null;
+  }
+}
+
+export default function PreferenceDetails() {
+  const { client } = useFrontier();
+  const { name } = useParams();
+
+  const [isPreferencesLoading, setIsPreferencesLoading] = useState(false);
+
+  const [trait, setTrait] = useState<V1Beta1PreferenceTrait | undefined>();
+  const [preference, setPreference] = useState<V1Beta1Preference | undefined>();
+  const [value, setValue] = useState("");
+  const [isActionLoading, setIsActionLoading] = useState(false);
+
+  const pageHeader = {
+    title: "Preference",
+    breadcrumb: [
+      {
+        href: `/preferences`,
+        name: `Preferences`,
+      },
+      {
+        href: `/preferences/${trait?.name}`,
+        name: `${trait?.title}`,
+      },
+    ],
+  };
+
+  useEffect(() => {
+    async function getPreference(name: string) {
```

Comment:
we can use useOutletContext for passing context to detail page.
For reference you can look into admin ui repo instead of fetching all preferences each time

Reply:
Created a Layout component and fetched data in that. the list and details component are children routes and read data form outlet context.